scipy/interpolate/fitpack/sproot.f

```diff
@@ -1,15 +1,19 @@
-      recursive subroutine sproot(t,n,c,zero,mest,m,ier)
+      recursive subroutine sproot(t,n,c,nc,k,zero,mest,m,ier)
```

Comment:
Actually, this subroutine does not need `k`, which is only used in `c` array size check in `fitpack.pyf`. But if the `k` is not added, this compile error happened:
```
scipy/interpolate/dfitpackmodule.c:1214:26: error: use of undeclared identifier 'k'
    CHECKARRAY(len(c)>=n-k-1,"len(c)>=n-k-1","2nd argument c") {
                         ^
1 error generated.
```
I'm a beginner of F2PY, does anyone have a good idea to remove the unnecessary argument addition?

Reply:
Since `sproot` is only implemented for cubic splines, `k=3`, we can use this fact. Two options: either just hardcode `check(len(c) >= n - 3 - 1)`, or use the `intent(hide)` qualifier, which hides the Fortran argument from the python interface:

```diff
diff --git a/scipy/interpolate/_fitpack_impl.py b/scipy/interpolate/_fitpack_impl.py
index 81f76175b5..6a05c9614a 100644
--- a/scipy/interpolate/_fitpack_impl.py
+++ b/scipy/interpolate/_fitpack_impl.py
@@ -714,7 +714,7 @@ def sproot(tck, mest=10):
     else:
         if len(t) < 8:
             raise TypeError("The number of knots %d>=8" % len(t))
-        z, m, ier = dfitpack.sproot(t, c, k, mest)
+        z, m, ier = dfitpack.sproot(t, c, mest)
         if ier == 10:
             raise TypeError("Invalid input data. "
                             "t1<=..<=t4<t5<..<tn-3<=..<=tn must hold.")
diff --git a/scipy/interpolate/src/fitpack.pyf b/scipy/interpolate/src/fitpack.pyf
index 966a9b3e87..5a7cca4dab 100644
--- a/scipy/interpolate/src/fitpack.pyf
+++ b/scipy/interpolate/src/fitpack.pyf
@@ -161,7 +161,7 @@ static F_INT calc_regrid_lwrk(F_INT mx, F_INT my, F_INT kx, F_INT ky,
        integer intent(hide),depend(t),check(n>=8) :: n=len(t)
        real*8 dimension(nc), intent(in), depend(n,k,t),check(len(c)>=n-k-1) :: c
        integer intent(hide), depend(c) :: nc=len(c)
-       integer intent(in) :: k
+       integer intent(hide) :: k=3
        real*8 dimension(mest),intent(out),depend(mest) :: zero
        integer optional,intent(in),depend(n) :: mest=3*(n-7)
        integer intent(out) :: m
```

Reply:
Thank you so much for your advice. I agree the hard-coded "check(len(c) >= n - 3 - 1)" is the best, and I added a comment for later reference.